packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -263,30 +282,61 @@ export class QueuedRequestController extends BaseController<
    */
   async #processNextBatch() {
     const firstRequest = this.#requestQueue.shift() as QueuedRequest;
-    this.#originOfCurrentBatch = firstRequest.origin;
-    this.#networkClientIdOfCurrentBatch = firstRequest.networkClientId;
-    const batch = [firstRequest.processRequest];
+    this.#originOfCurrentBatch = firstRequest.request.origin;
+    this.#networkClientIdOfCurrentBatch = firstRequest.request.networkClientId;
+    const batch = [firstRequest];
+
+    let networkSwitchError: unknown;
+    try {
+      // If globally selected network is different from origin selected network,
+      // switch network before processing batch
+      await this.#switchNetworkIfNecessary(
```

Comment:
What does moving this call effectively do and do we have a test for this change in particular?

Reply:
This function used to build a batch, switches the chain if necessary, then processes the batch.

With this change, the order of the first two steps has been reversed. We switch before building out the rest of the batch.

The order of these two steps doesn't matter functionally because the batch doesn't get used for anything until the last step, the processing step. The switch was moved up here because in the "can request switch without approval" case, we want to skip the "build a batch" step. We want to treat the single request as a batch in that case.

Preserving the previous order would have meant duplicating this code, so probably better to just move it ahead here. No test is relevant because this isn't associated with a functional change.

Reply:
nothing changes for this test by moving this block before the `while` batch gathering loop. I moved it because this call needs to happen before processing any request in the batch and my new changes handle the first request outside of the `for` batch processing loop